keps/sig-storage/1979-object-storage-support/README.md

```diff
@@ -0,0 +1,730 @@
+
+For greenfield, COSI creates the `Bucket` based on values in the `BucketRequest` and `BucketClass`. For brownfield, an admin manually creates the `Bucket`, filling in BucketClass fields, such as `allowedNamespaces`. COSI populates fields returned by the provisioner, and binds the `Bucket` to the `BucketAccess`.
+
+Since there is a 1:1 mapping between a BR and a `Bucket`, when multiple BRs request access to the same backend bucket, each separate `Bucket` points to the same backend bucket.
+
+```yaml
+apiVersion: cosi.io/v1alpha1
+kind: Bucket
+metadata:
+  name: [1]
+  labels: [2]
+    cosi.io/provisioner:
+    cosi.io/bucket-prefix:
+  finalizers:
+    - cosi.io/finalizer [3]
+spec:
+  provisioner: [4]
+  retentionPolicy: [5]
+  anonymousAccessMode: [6]
+  bucketClassName: [7]
+  bucketRequest: [8]
+  allowedNamespaces: [9]
+    - name:
+  protocol: [10]
+    name:
+    version:
+    azureBlob:
+      containerName:
+      storageAccount:
+    s3:
+      endpoint:
+      bucketName:
+      region:
+      signatureVersion:
+    gs:
+      bucketName:
+      privateKeyName:
+      projectId:
+      serviceAccount:
+  parameters: [11]
+status:
+  bucketAvailable: [12]
+```
+
+1. `name`: When created by COSI, the `Bucket` name is generated in this format: _"<BR.namespace*>-<BR.name*>-<uuid>"_. If an admin creates a `Bucket`, as is necessary for brownfield access, they can use any name. The uuid is unique within a cluster. `*` the first 100 characters of the namespace and the name are used.
+    cosi.io/provisioner:
+    cosi.io/bucket-prefix:
+2. `labels`: added by COSI. The "cosi.io/provisioner" key's value is the provisioner name. The "cosi.io/bucket-prefix" key's value is the `BucketRequest.bucketPrefix` value when specified. Characters that do not adhere to [Kubernetes label conventions](https://kubernetes.io/docs/concepts/overview/working-with-objects/labels/#syntax-and-character-set) will be converted to ‘-’.
+3. `finalizers`: added by COSI to defer `Bucket` deletion until backend deletion ops succeed. Implementation may add one finalizer for the BR and one for the BA.
+4. `provisioner`: (optional) The provisioner field as defined in the `BucketClass`. If empty then there is no driver/sidecar, thus the admin had to create the `Bucket` and `BucketAccess` instances.
+5. `retentionPolicy`: Prescribes the outcome when the `BucketRequest` is deleted.
+   - _Retain_: (default) the `Bucket` and its data are preserved. The `Bucket` can potentially be reused.
+   - _Delete_: the bucket and its contents are destroyed.
+> Note: the `Bucket`'s retention policy is set to "Retain" as a default. Exercise caution when using the "Delete" retention policy as the bucket content will be deleted.
+> Note: a `Bucket` is not deleted if it is bound to a `BucketRequest` or `BucketAccess`.
```

Comment:
This may contradictory [Delete Bucket](#DeleteBucket) section, "Delete bucket his means that even if a `BucketAccess` references a `Bucket`, the "Bucket is still deleted and the BA is orphaned"

Reply:
yeah, your understanding is correct. I'm not sure if you're asking a question, or just explaining what you're understanding

Reply:
> ...to a `BucketRequest` or `BucketAccess`.

the "or `BucketAccess` is wrong and was missed in earlier edits"

Reply:
Yeah it was not a question, just pointed out two different statements